extensions/amp-analytics/0.1/linker-manager.js

```diff
@@ -308,19 +304,43 @@ export class LinkerManager {
 
 
   /**
-   * Add the linker pairs as <input> elements to form.
+   * Add the linker data to form. If action-xhr is present we can set a new
+   * linker-xhr to use, if not we fallback to adding hidden inputs.
    * @param {!Element} form
    * @param {string} linkerName
    */
   addDataToForm_(form, linkerName) {
-    if (!this.resolvedLinkers_[linkerName]) {
+    const linkerValue = this.resolvedLinkers_[linkerName];
+    if (!linkerValue) {
       return;
     }
 
+    // Runtime controls submits with `action-xhr`, so we can point it at the new
+    // url set as linker-xhr attribute.
+    const actionXhrUrl = form.getAttribute('action-xhr');
```

Comment:
the doc says it's method=GET can also have `action-xhr`.
https://github.com/ampproject/amphtml/blob/master/extensions/amp-form/amp-form.md#action

Reply:
discussed offline, it is correct that when action-xhr is in use we should always append URL